packages/flutter/lib/src/widgets/overflow_bar.dart

```diff
@@ -0,0 +1,489 @@
+  }
+}
+
+class _OverflowBarParentData extends ContainerBoxParentData<RenderBox> { }
+
+class _RenderOverflowBar extends RenderBox
+    with ContainerRenderObjectMixin<RenderBox, _OverflowBarParentData>,
+         RenderBoxContainerDefaultsMixin<RenderBox, _OverflowBarParentData> {
+  _RenderOverflowBar({
+    List<RenderBox> children,
+    double spacing = 0.0,
+    double overflowSpacing = 0.0,
+    OverflowBarAlignment overflowAlignment = OverflowBarAlignment.start,
+    VerticalDirection overflowDirection = VerticalDirection.down,
+    TextDirection textDirection,
+    Clip clipBehavior = Clip.none,
+  }) : assert(spacing != null),
+       assert(overflowSpacing != null),
+       assert(overflowAlignment != null),
+       assert(textDirection != null),
+       assert(clipBehavior != null),
+       _spacing = spacing,
+       _overflowSpacing = overflowSpacing,
+       _overflowAlignment = overflowAlignment,
+       _overflowDirection = overflowDirection,
+       _textDirection = textDirection,
+       _clipBehavior = clipBehavior {
+    addAll(children);
+  }
+
+  double get spacing => _spacing;
+  double _spacing;
+  set spacing (double value) {
+    assert(value != null);
+    if (_spacing == value)
+      return;
+    _spacing = value;
+    markNeedsLayout();
+  }
+
+  double get overflowSpacing => _overflowSpacing;
+  double _overflowSpacing;
+  set overflowSpacing (double value) {
+    assert(value != null);
+    if (_overflowSpacing == value)
+      return;
+    _overflowSpacing = value;
+    markNeedsLayout();
+  }
+
+  OverflowBarAlignment get overflowAlignment => _overflowAlignment;
+  OverflowBarAlignment _overflowAlignment;
+  set overflowAlignment (OverflowBarAlignment value) {
+    assert(value != null);
+    if (_overflowAlignment == value)
+      return;
+    _overflowAlignment = value;
+    markNeedsLayout();
+  }
+
+  VerticalDirection get overflowDirection => _overflowDirection;
+  VerticalDirection _overflowDirection;
+  set overflowDirection (VerticalDirection value) {
+    assert(value != null);
+    if (_overflowDirection == value)
+      return;
+    _overflowDirection = value;
+    markNeedsLayout();
+  }
+
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection == value)
+      return;
+    _textDirection = value;
+    markNeedsLayout();
+  }
+
+  Clip get clipBehavior => _clipBehavior;
+  Clip _clipBehavior = Clip.none;
+  set clipBehavior(Clip value) {
+    assert(value != null);
+    if (value == _clipBehavior)
+      return;
+    _clipBehavior = value;
+    markNeedsPaint();
+    markNeedsSemanticsUpdate();
+  }
+
+  @override
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! _OverflowBarParentData)
+      child.parentData = _OverflowBarParentData();
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    RenderBox child = firstChild;
```

Comment:
Is this calculation correct? I'd assumed that if width is big enough everything gets layed out in a row and then the hight would not be the sum of the children heights, no?

Reply:
Yes, you are correct, this method is incorrect. The min intrinsic height should be: if the sum of child intrinsic widths fit, then the max child height, otherwise the sum of the child heights.

Reply:
I've fixed this however I still need to add some tests for min/max intrinsic height.